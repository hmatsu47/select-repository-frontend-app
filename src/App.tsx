import { Component } from "solid-js";
import Box from "@suid/material/Box";
import Stack from "@suid/material/Stack";
import { TitleBar } from "./TitleBar";

export const App: Component = () => {
  return (
    <>
      <TitleBar />
      <Box
        sx={{
          padding: "10px 0 0 0",
          width: "100%",
          minWidth: "960px",
          display: "flex",
          justifyContent: "center",
        }}
        aria-live="polite"
      >
        <Stack spacing={2} direction="column"></Stack>
      </Box>
    </>
  );
};
