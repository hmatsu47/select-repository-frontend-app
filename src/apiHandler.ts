import { fetchWithTimeout } from "./fetchWithTimeout";

// APIからデータ取得（GET）
export const getApiData = async (url: string) => {
  return await (await fetchWithTimeout(url)).json();
};
// APIからデータ取得（POST）
export const postApiData = async (url: string, object: Object) => {
  return await (await fetchWithTimeout(url, object)).json();
};
