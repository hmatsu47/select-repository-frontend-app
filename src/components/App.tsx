import { Component } from "solid-js";
import Box from "@suid/material/Box";
import Stack from "@suid/material/Stack";
import { ImageList } from "./ImageList";
import { LastReleased } from "./LastReleased";
import { Message } from "./Message";
import { ReleaseSetting } from "./ReleaseSetting";
import { RepositorySelector } from "./RepositorySelector";
import { ServiceSelector } from "./ServiceSelector";
import { TitleBar } from "./TitleBar";

export const App: Component = () => {
  return (
    <>
      <TitleBar />
      <Box
        sx={{
          padding: "10px 10px 0 10px",
          width: "100%",
          minWidth: "1024px",
          display: "flex",
          // justifyContent: "center",
        }}
        aria-live="polite"
      >
        <Stack spacing={2} direction="column">
          <ServiceSelector />
          <LastReleased />
          <RepositorySelector />
          <ReleaseSetting />
          <Message />
          <ImageList />
        </Stack>
      </Box>
    </>
  );
};
