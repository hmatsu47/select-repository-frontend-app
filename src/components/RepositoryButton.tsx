import Button from "@suid/material/Button";
import { fetchImages } from "../api/fetchImages";
import { repository, setRepository } from "../signal";
import { RepositoryItem } from "../type";

type Props = {
  repositoryItem: RepositoryItem;
};

export const RepositoryButton = (props: Props) => {
  return (
    <Button
      variant={
        repository() === props.repositoryItem.name ? "contained" : "outlined"
      }
      size="small"
      color={repository() === props.repositoryItem.name ? "primary" : "inherit"}
      onClick={async (e) => {
        setRepository(props.repositoryItem.name);
        await fetchImages();
      }}
      sx={{ textTransform: "none" }}
    >
      {props.repositoryItem.name}
    </Button>
  );
};
