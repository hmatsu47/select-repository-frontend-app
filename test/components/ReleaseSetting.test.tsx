import { describe, expect, test, vi } from "vitest";
import { fireEvent, render } from "solid-testing-library";
import { formatSnapshot } from "../common/formatSnapshot";
import { ReleaseSetting } from "../../src/components/ReleaseSetting";
import {
  releaseAt,
  setImages,
  setImageUri,
  setIsReleased,
  setIsReleaseSelected,
  setReleaseAt,
  setService,
  isCancel,
  isOpenedConfirm,
  setIsCancel,
  setIsOpenedConfirm,
} from "../../src/signal";
import { ImageItem } from "../../src/type";

describe("<ReleaseSetting />", () => {
  beforeEach(() => {
    // フェイクタイマーを使う
    vi.useFakeTimers();
    const date = new Date(2022, 0, 1, 0, 0);
    vi.setSystemTime(date);
  });
  afterEach(() => {
    // 正しいタイマーに戻す
    vi.useRealTimers();
  });
  const buttonList = [
    {
      title: "任意のリリース時間をセット",
      buttonTitle: null,
      text: null,
      expected: new Date(2022, 8, 11, 21, 1),
    },
    {
      title: "翌朝リリースするボタンをクリック",
      buttonTitle: "翌朝リリースする",
      text: "翌朝 04:05",
      expected: new Date(2022, 0, 2, 4, 5),
    },
    {
      title: "即時リリースするボタンをクリック",
      buttonTitle: "即時リリースする",
      text: "現在（即時リリースする）",
      expected: new Date(2022, 0, 1, 0, 0),
    },
  ];
  buttonList.forEach((testCase) => {
    test(testCase.title, async () => {
      setIsReleased(false);
      setImageUri(undefined);
      setReleaseAt(undefined);
      setIsReleaseSelected(true); // false のテストケースが入ると失敗するので true のみ
      setIsCancel(false);
      setService("service1");
      setImages([
        {
          service: "service1",
          repository: "hoge",
          digest:
            "sha256:4c716d4cf211c7b7d2f3233c941771ad0507ea5bacf93b492766aa41ae9f720d",
          pushed_at: new Date(),
          repository_name: "hoge",
          size: 10000,
          tags: ["latest", "hogera"],
          uri: "000000000000.dkr.ecr.ap-northeast-1.amazonaws.com/hoge:latest",
        } as ImageItem,
      ]);
      const { container, findByText, findByTitle, unmount } = render(() => (
        <ReleaseSetting />
      ));
      const expectedReleaseUri = (await findByText(
        "次回リリースイメージ URI"
      )) as HTMLElement;
      expect(expectedReleaseUri).toHaveTextContent("次回リリースイメージ URI");
      const expectedReleaseAt = (await findByText(
        "次回リリース日時"
      )) as HTMLElement;
      expect(expectedReleaseAt).toHaveTextContent("次回リリース日時");
      // css の名前が動的に変わるので固定値に置換
      const htmlBefore = formatSnapshot(container.innerHTML);
      expect(htmlBefore).toMatchSnapshot();
      if (testCase.text) {
        // ボタンクリック
        const button = (await findByTitle(
          testCase.buttonTitle
        )) as HTMLInputElement;
        expect(button).toHaveTextContent(testCase.text);
        fireEvent.click(button);
        expect(releaseAt()).toStrictEqual(testCase.expected);
      } else {
        // 日付指定
        setReleaseAt(testCase.expected);
      }
      // 日付指定後のスナップショット
      const htmlAfterTimeSet = formatSnapshot(container.innerHTML);
      expect(htmlAfterTimeSet).toMatchSnapshot();
      // イメージ URI 指定
      const dummyUri =
        "000000000000.dkr.ecr.ap-northeast-1.amazonaws.com/hoge:latest";
      setImageUri(dummyUri);
      const expectedReleaseUriValue = (await findByText(
        dummyUri
      )) as HTMLElement;
      expect(expectedReleaseUriValue).toHaveTextContent(dummyUri);
      // イメージ URI 指定後のスナップショット
      const htmlAfterUriSet = formatSnapshot(container.innerHTML);
      expect(htmlAfterUriSet).toMatchSnapshot();
      // 次回リリースをセット
      const buttonReleaseSet = (await findByTitle(
        "次回リリースをセット"
      )) as HTMLInputElement;
      fireEvent.click(buttonReleaseSet);
      expect(buttonReleaseSet).toHaveTextContent(
        "指定のイメージ URI と日時をセット"
      );
      expect(isCancel()).toBe(false);
      expect(isOpenedConfirm()).toBe(true);
      // 次回リリースを取り消し
      setIsOpenedConfirm(false);
      const buttonReleaseReset = (await findByTitle(
        "次回リリースを取り消し"
      )) as HTMLInputElement;
      fireEvent.click(buttonReleaseReset);
      expect(buttonReleaseReset).toHaveTextContent("リリースを取り消し");
      expect(isCancel()).toBe(true);
      expect(isOpenedConfirm()).toBe(true);
      unmount();
    });
  });
});
