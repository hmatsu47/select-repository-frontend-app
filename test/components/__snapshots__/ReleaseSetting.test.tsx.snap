// Vitest Snapshot v1

exports[`<ReleaseSetting /> > 任意のリリース時間をセット 1`] = `"<div class=\\"MuiBox-root css-xxxxxxxx\\" aria-live=\\"polite\\"><div class=\\"MuiStack-root css-xxxxxxxx\\"><div class=\\"MuiStack-root css-xxxxxxxx\\"><h6 class=\\"MuiTypography-root MuiTypography-h6 MuiTypography-root css-xxxxxxxx\\">次回リリースイメージ URI</h6><h6 class=\\"MuiTypography-root MuiTypography-subtitle1 MuiTypography-root css-xxxxxxxx\\">下のリストから選択してください</h6></div><div class=\\"MuiStack-root css-xxxxxxxx\\"><h6 class=\\"MuiTypography-root MuiTypography-h6 MuiTypography-root css-xxxxxxxx\\"><label id=\\"releaseAt\\">次回リリース日時</label></h6><input type=\\"datetime-local\\" aria-labelledby=\\"releaseAt\\" max=\\"9999-12-31 23:59\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"翌朝リリースする\\" type=\\"button\\">翌朝 04:05<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"即時リリースする\\" type=\\"button\\">現在（即時リリースする）<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></div><div class=\\"MuiStack-root css-xxxxxxxx\\"><h6 class=\\"MuiTypography-root MuiTypography-h6 MuiTypography-root css-xxxxxxxx\\">次回リリースセット</h6><button class=\\"MuiButtonBase-root Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"-1\\" title=\\"次回リリースをセット\\" type=\\"button\\" disabled=\\"\\">指定のイメージ URI と日時をセット</button><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"次回リリースを取り消し\\" type=\\"button\\">リリースを取り消し<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></div></div></div>"`;

exports[`<ReleaseSetting /> > 任意のリリース時間をセット 2`] = `"<div class=\\"MuiBox-root css-xxxxxxxx\\" aria-live=\\"polite\\"><div class=\\"MuiStack-root css-xxxxxxxx\\"><div class=\\"MuiStack-root css-xxxxxxxx\\"><h6 class=\\"MuiTypography-root MuiTypography-h6 MuiTypography-root css-xxxxxxxx\\">次回リリースイメージ URI</h6><h6 class=\\"MuiTypography-root MuiTypography-subtitle1 MuiTypography-root css-xxxxxxxx\\">下のリストから選択してください</h6></div><div class=\\"MuiStack-root css-xxxxxxxx\\"><h6 class=\\"MuiTypography-root MuiTypography-h6 MuiTypography-root css-xxxxxxxx\\"><label id=\\"releaseAt\\">次回リリース日時</label></h6><input type=\\"datetime-local\\" aria-labelledby=\\"releaseAt\\" max=\\"9999-12-31 23:59\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"翌朝リリースする\\" type=\\"button\\">翌朝 04:05<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"即時リリースする\\" type=\\"button\\">現在（即時リリースする）<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></div><div class=\\"MuiStack-root css-xxxxxxxx\\"><h6 class=\\"MuiTypography-root MuiTypography-h6 MuiTypography-root css-xxxxxxxx\\">次回リリースセット</h6><button class=\\"MuiButtonBase-root Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"-1\\" title=\\"次回リリースをセット\\" type=\\"button\\" disabled=\\"\\">指定のイメージ URI と日時をセット</button><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"次回リリースを取り消し\\" type=\\"button\\">リリースを取り消し<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></div></div></div>"`;

exports[`<ReleaseSetting /> > 任意のリリース時間をセット 3`] = `"<div class=\\"MuiBox-root css-xxxxxxxx\\" aria-live=\\"polite\\"><div class=\\"MuiStack-root css-xxxxxxxx\\"><div class=\\"MuiStack-root css-xxxxxxxx\\"><h6 class=\\"MuiTypography-root MuiTypography-h6 MuiTypography-root css-xxxxxxxx\\">次回リリースイメージ URI</h6><h6 class=\\"MuiTypography-root MuiTypography-subtitle1 MuiTypography-root css-xxxxxxxx\\">000000000000.dkr.ecr.ap-northeast-1.amazonaws.com/hoge:latest</h6></div><div class=\\"MuiStack-root css-xxxxxxxx\\"><h6 class=\\"MuiTypography-root MuiTypography-h6 MuiTypography-root css-xxxxxxxx\\"><label id=\\"releaseAt\\">次回リリース日時</label></h6><input type=\\"datetime-local\\" aria-labelledby=\\"releaseAt\\" max=\\"9999-12-31 23:59\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"翌朝リリースする\\" type=\\"button\\">翌朝 04:05<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"即時リリースする\\" type=\\"button\\">現在（即時リリースする）<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></div><div class=\\"MuiStack-root css-xxxxxxxx\\"><h6 class=\\"MuiTypography-root MuiTypography-h6 MuiTypography-root css-xxxxxxxx\\">次回リリースセット</h6><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"次回リリースをセット\\" type=\\"button\\">指定のイメージ URI と日時をセット<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"次回リリースを取り消し\\" type=\\"button\\">リリースを取り消し<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></div></div></div>"`;

exports[`<ReleaseSetting /> > 即時リリースするボタンをクリック 1`] = `"<div class=\\"MuiBox-root css-xxxxxxxx\\" aria-live=\\"polite\\"><div class=\\"MuiStack-root css-xxxxxxxx\\"><div class=\\"MuiStack-root css-xxxxxxxx\\"><h6 class=\\"MuiTypography-root MuiTypography-h6 MuiTypography-root css-xxxxxxxx\\">次回リリースイメージ URI</h6><h6 class=\\"MuiTypography-root MuiTypography-subtitle1 MuiTypography-root css-xxxxxxxx\\">下のリストから選択してください</h6></div><div class=\\"MuiStack-root css-xxxxxxxx\\"><h6 class=\\"MuiTypography-root MuiTypography-h6 MuiTypography-root css-xxxxxxxx\\"><label id=\\"releaseAt\\">次回リリース日時</label></h6><input type=\\"datetime-local\\" aria-labelledby=\\"releaseAt\\" max=\\"9999-12-31 23:59\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"翌朝リリースする\\" type=\\"button\\">翌朝 04:05<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"即時リリースする\\" type=\\"button\\">現在（即時リリースする）<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></div><div class=\\"MuiStack-root css-xxxxxxxx\\"><h6 class=\\"MuiTypography-root MuiTypography-h6 MuiTypography-root css-xxxxxxxx\\">次回リリースセット</h6><button class=\\"MuiButtonBase-root Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"-1\\" title=\\"次回リリースをセット\\" type=\\"button\\" disabled=\\"\\">指定のイメージ URI と日時をセット</button><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"次回リリースを取り消し\\" type=\\"button\\">リリースを取り消し<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></div></div></div>"`;

exports[`<ReleaseSetting /> > 即時リリースするボタンをクリック 2`] = `"<div class=\\"MuiBox-root css-xxxxxxxx\\" aria-live=\\"polite\\"><div class=\\"MuiStack-root css-xxxxxxxx\\"><div class=\\"MuiStack-root css-xxxxxxxx\\"><h6 class=\\"MuiTypography-root MuiTypography-h6 MuiTypography-root css-xxxxxxxx\\">次回リリースイメージ URI</h6><h6 class=\\"MuiTypography-root MuiTypography-subtitle1 MuiTypography-root css-xxxxxxxx\\">下のリストから選択してください</h6></div><div class=\\"MuiStack-root css-xxxxxxxx\\"><h6 class=\\"MuiTypography-root MuiTypography-h6 MuiTypography-root css-xxxxxxxx\\"><label id=\\"releaseAt\\">次回リリース日時</label></h6><input type=\\"datetime-local\\" aria-labelledby=\\"releaseAt\\" max=\\"9999-12-31 23:59\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"翌朝リリースする\\" type=\\"button\\">翌朝 04:05<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"即時リリースする\\" type=\\"button\\">現在（即時リリースする）<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></div><div class=\\"MuiStack-root css-xxxxxxxx\\"><h6 class=\\"MuiTypography-root MuiTypography-h6 MuiTypography-root css-xxxxxxxx\\">次回リリースセット</h6><button class=\\"MuiButtonBase-root Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"-1\\" title=\\"次回リリースをセット\\" type=\\"button\\" disabled=\\"\\">指定のイメージ URI と日時をセット</button><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"次回リリースを取り消し\\" type=\\"button\\">リリースを取り消し<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></div></div></div>"`;

exports[`<ReleaseSetting /> > 即時リリースするボタンをクリック 3`] = `"<div class=\\"MuiBox-root css-xxxxxxxx\\" aria-live=\\"polite\\"><div class=\\"MuiStack-root css-xxxxxxxx\\"><div class=\\"MuiStack-root css-xxxxxxxx\\"><h6 class=\\"MuiTypography-root MuiTypography-h6 MuiTypography-root css-xxxxxxxx\\">次回リリースイメージ URI</h6><h6 class=\\"MuiTypography-root MuiTypography-subtitle1 MuiTypography-root css-xxxxxxxx\\">000000000000.dkr.ecr.ap-northeast-1.amazonaws.com/hoge:latest</h6></div><div class=\\"MuiStack-root css-xxxxxxxx\\"><h6 class=\\"MuiTypography-root MuiTypography-h6 MuiTypography-root css-xxxxxxxx\\"><label id=\\"releaseAt\\">次回リリース日時</label></h6><input type=\\"datetime-local\\" aria-labelledby=\\"releaseAt\\" max=\\"9999-12-31 23:59\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"翌朝リリースする\\" type=\\"button\\">翌朝 04:05<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"即時リリースする\\" type=\\"button\\">現在（即時リリースする）<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></div><div class=\\"MuiStack-root css-xxxxxxxx\\"><h6 class=\\"MuiTypography-root MuiTypography-h6 MuiTypography-root css-xxxxxxxx\\">次回リリースセット</h6><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"次回リリースをセット\\" type=\\"button\\">指定のイメージ URI と日時をセット<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"次回リリースを取り消し\\" type=\\"button\\">リリースを取り消し<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></div></div></div>"`;

exports[`<ReleaseSetting /> > 翌朝リリースするボタンをクリック 1`] = `"<div class=\\"MuiBox-root css-xxxxxxxx\\" aria-live=\\"polite\\"><div class=\\"MuiStack-root css-xxxxxxxx\\"><div class=\\"MuiStack-root css-xxxxxxxx\\"><h6 class=\\"MuiTypography-root MuiTypography-h6 MuiTypography-root css-xxxxxxxx\\">次回リリースイメージ URI</h6><h6 class=\\"MuiTypography-root MuiTypography-subtitle1 MuiTypography-root css-xxxxxxxx\\">下のリストから選択してください</h6></div><div class=\\"MuiStack-root css-xxxxxxxx\\"><h6 class=\\"MuiTypography-root MuiTypography-h6 MuiTypography-root css-xxxxxxxx\\"><label id=\\"releaseAt\\">次回リリース日時</label></h6><input type=\\"datetime-local\\" aria-labelledby=\\"releaseAt\\" max=\\"9999-12-31 23:59\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"翌朝リリースする\\" type=\\"button\\">翌朝 04:05<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"即時リリースする\\" type=\\"button\\">現在（即時リリースする）<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></div><div class=\\"MuiStack-root css-xxxxxxxx\\"><h6 class=\\"MuiTypography-root MuiTypography-h6 MuiTypography-root css-xxxxxxxx\\">次回リリースセット</h6><button class=\\"MuiButtonBase-root Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"-1\\" title=\\"次回リリースをセット\\" type=\\"button\\" disabled=\\"\\">指定のイメージ URI と日時をセット</button><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"次回リリースを取り消し\\" type=\\"button\\">リリースを取り消し<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></div></div></div>"`;

exports[`<ReleaseSetting /> > 翌朝リリースするボタンをクリック 2`] = `"<div class=\\"MuiBox-root css-xxxxxxxx\\" aria-live=\\"polite\\"><div class=\\"MuiStack-root css-xxxxxxxx\\"><div class=\\"MuiStack-root css-xxxxxxxx\\"><h6 class=\\"MuiTypography-root MuiTypography-h6 MuiTypography-root css-xxxxxxxx\\">次回リリースイメージ URI</h6><h6 class=\\"MuiTypography-root MuiTypography-subtitle1 MuiTypography-root css-xxxxxxxx\\">下のリストから選択してください</h6></div><div class=\\"MuiStack-root css-xxxxxxxx\\"><h6 class=\\"MuiTypography-root MuiTypography-h6 MuiTypography-root css-xxxxxxxx\\"><label id=\\"releaseAt\\">次回リリース日時</label></h6><input type=\\"datetime-local\\" aria-labelledby=\\"releaseAt\\" max=\\"9999-12-31 23:59\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"翌朝リリースする\\" type=\\"button\\">翌朝 04:05<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"即時リリースする\\" type=\\"button\\">現在（即時リリースする）<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></div><div class=\\"MuiStack-root css-xxxxxxxx\\"><h6 class=\\"MuiTypography-root MuiTypography-h6 MuiTypography-root css-xxxxxxxx\\">次回リリースセット</h6><button class=\\"MuiButtonBase-root Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"-1\\" title=\\"次回リリースをセット\\" type=\\"button\\" disabled=\\"\\">指定のイメージ URI と日時をセット</button><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"次回リリースを取り消し\\" type=\\"button\\">リリースを取り消し<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></div></div></div>"`;

exports[`<ReleaseSetting /> > 翌朝リリースするボタンをクリック 3`] = `"<div class=\\"MuiBox-root css-xxxxxxxx\\" aria-live=\\"polite\\"><div class=\\"MuiStack-root css-xxxxxxxx\\"><div class=\\"MuiStack-root css-xxxxxxxx\\"><h6 class=\\"MuiTypography-root MuiTypography-h6 MuiTypography-root css-xxxxxxxx\\">次回リリースイメージ URI</h6><h6 class=\\"MuiTypography-root MuiTypography-subtitle1 MuiTypography-root css-xxxxxxxx\\">000000000000.dkr.ecr.ap-northeast-1.amazonaws.com/hoge:latest</h6></div><div class=\\"MuiStack-root css-xxxxxxxx\\"><h6 class=\\"MuiTypography-root MuiTypography-h6 MuiTypography-root css-xxxxxxxx\\"><label id=\\"releaseAt\\">次回リリース日時</label></h6><input type=\\"datetime-local\\" aria-labelledby=\\"releaseAt\\" max=\\"9999-12-31 23:59\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"翌朝リリースする\\" type=\\"button\\">翌朝 04:05<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"即時リリースする\\" type=\\"button\\">現在（即時リリースする）<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></div><div class=\\"MuiStack-root css-xxxxxxxx\\"><h6 class=\\"MuiTypography-root MuiTypography-h6 MuiTypography-root css-xxxxxxxx\\">次回リリースセット</h6><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"次回リリースをセット\\" type=\\"button\\">指定のイメージ URI と日時をセット<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"次回リリースを取り消し\\" type=\\"button\\">リリースを取り消し<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></div></div></div>"`;
